pywinauto/Linux/keyboard.py

```diff
@@ -0,0 +1,547 @@
+    '<' : "less",
+    '>' : "greater",
+    '?' : "question",
+    '@' : "at",
+    '[' : "bracketleft",
+    ']' : "bracketright",
+    '\\' : "backslash",
+    '^' : "asciicircum",
+    '_' : "underscore",
+    '`' : "grave",
+    '{' : "braceleft",
+    '|' : "bar",
+    '}' : "braceright",
+    '~' : "asciitilde"
+}
+
+def _to_keycode(key):
+    """return python X11 keycode of symbol"""
+    return _display.keysym_to_keycode(Xlib.XK.string_to_keysym(key))
+
+INPUT_KEYBOARD = 1
+KEYEVENTF_EXTENDEDKEY = 1
+KEYEVENTF_KEYUP       = 2
+KEYEVENTF_UNICODE     = 4
+KEYEVENTF_SCANCODE    = 8
+VK_SHIFT        = _to_keycode(('Shift_L'))
+VK_CONTROL      = _to_keycode(('Control_L'))
+VK_MENU         = _to_keycode(('Menu'))
+
+# 'codes' recognized as {CODE( repeat)?}
+
+
+CODES = {
+    'BACK':     _to_keycode(('BackSpace')),
+    'BACKSPACE': _to_keycode(('BackSpace')),
+    'BKSP':     _to_keycode(('BackSpace')),
+    'BREAK':    _to_keycode(('Break')),
+    'BS':       _to_keycode(('BackSpace')),
+    'CAP':      _to_keycode(('Caps_Lock')),
+    'CAPSLOCK': _to_keycode(('Caps_Lock')),
+    'DEL':      _to_keycode(('Delete')),
+    'DELETE':   _to_keycode(('Delete')),
+    'DOWN':     _to_keycode(('Down')),
+    'END':      _to_keycode(('End')),
+    'ENTER':    _to_keycode(('Return')),
+    'ESC':      _to_keycode(('Escape')),
+    'F1':       _to_keycode(('F1')),
+    'F2':       _to_keycode(('F2')),
+    'F3':       _to_keycode(('F3')),
+    'F4':       _to_keycode(('F4')),
+    'F5':       _to_keycode(('F5')),
+    'F6':       _to_keycode(('F6')),
+    'F7':       _to_keycode(('F7')),
+    'F8':       _to_keycode(('F8')),
+    'F9':       _to_keycode(('F9')),
+    'F10':      _to_keycode(('F10')),
+    'F11':      _to_keycode(('F11')),
+    'F12':      _to_keycode(('F12')),
+    'F13':      _to_keycode(('F13')),
+    'F14':      _to_keycode(('F14')),
+    'F15':      _to_keycode(('F15')),
+    'F16':      _to_keycode(('F16')),
+    'F17':      _to_keycode(('F17')),
+    'F18':      _to_keycode(('F18')),
+    'F19':      _to_keycode(('F19')),
+    'F20':      _to_keycode(('F20')),
+    'F21':      _to_keycode(('F21')),
+    'F22':      _to_keycode(('F22')),
+    'F23':      _to_keycode(('F23')),
+    'F24':      _to_keycode(('F24')),
+    'HELP':     _to_keycode(('Help')),
+    'HOME':     _to_keycode(('Home')),
+    'INS':      _to_keycode(('Insert')),
+    'INSERT':   _to_keycode(('Insert')),
+    'LEFT':     _to_keycode(('Left')),
+    'LWIN':     _to_keycode(('Super_L')),
+    'NUMLOCK':  _to_keycode(('Num_Lock')),
+    'PGDN':     _to_keycode(('Page_Down')),
+    'PGUP':     _to_keycode(('Page_Up')),
+    'PRTSC':    _to_keycode(('Print')),
+    'RIGHT':    _to_keycode(('Right')),
+    'RMENU':    _to_keycode(('Alt_R')),
+    'RWIN':     _to_keycode(('Super_R')),
+    'SCROLLLOCK': _to_keycode(('Scroll_Lock')),
+    'SPACE':     _to_keycode(('space')),
+    'TAB':       _to_keycode(('Tab')),
+    'UP':        _to_keycode(('Up')),
```

Comment:
Why brackets are doubled? It's not a tuple, because tuple should contain at least 1 comma: `('Up', )`. But this duplication looks strange.